### ISSUE TITLE
Enhancement: static asset separation

### ISSUE THREAD

**Issue by corrideat** [roundcube/roundcubemail#5738]
Roundcube currently comes in a package that includes some PHP code and some static assets. Additional plugins and themes also follow this structure.

I believe it would be useful to provide some separation between these. This could allow for the following:

1. Storing Roundcube's executable files from outside the web root. This can help, for example, for easier file permission management and reduces the risk of exposing sensitive information (e.g., database configuration, passwords, etc.) in misconfigured or misbehaved servers.
2. With an additional configuration option (base resource path), much as it is currently done for the logo image, static assets could be served from a different domain or a CDN. 

Having a look, I don't believe this should be too difficult to implement in code, although it can be a greater issue ensuring that themes and plugins' files remain accessible (as, basically, they'd need to place files in two locations instead of a single directory, as it is now.)

In addition, a(n optional) [subresource integrity](https://w3c.github.io/webappsec-subresource-integrity/) tag can be added to the code generating HTML elements to prevent potential vulnerabilities that could arise from implementing this feature (such as a malicious CDN.)

**Comment by alecpl** [comment]
We have already assets_path and assets_dir settings. So, you can separate assets, but you have to do it by your own. We have also public_html/ folder. The only issue is with plugins that as you noticed would need to split resources too, and I don't really like this idea.

BTW, Kolab already does this on a packaging level.

**Comment by corrideat** [comment]
Yes, while `assets_path` kind of allows separation, the manual process is a bit tedious and error-prone. (I have written a simple script to catch most files, by extension, but it won't work if some plugin uses an extension I didn't think of.)

Hence, my idea was more along the lines of having two main directories, say `application` (code) and `resources` (static files). Then, the process for implemeting separation is server-side is very simple.

For plugins, I'm thinking a compromise would be to recommend this same separation _within the plugin directory_. Then, the files would still need to be manually copied, but at least won't require hunting down the right files.

**Comment by alecpl** [comment]
#9294 has been merged. Now all resources are served by public_html/static.php. Imo, a separation is not needed.

**Comment by ner00** [comment]
> Now all resources are served by public_html/static.php.

Not all resources, apparently.

For example, when I test the `newmail_notifier` plugin in the Settings, to test notification popup or test notification sound, I get a 403 error because the plugin is redirecting the user directly to the original path of the resource, bypassing `static.php`.

Here's an excerpt from `newmail_notifier.js`:
```js
// Sound notification
function newmail_notifier_sound() {
    var src = rcmail.assets_path('plugins/newmail_notifier/sound');

    (new Audio(src + '.mp3')).play()
        .catch(function () {
            // fallback to the wav format
            (new Audio(src + '.wav')).play();
        });
}
```

... but I can make it work if I change the path to start with `static.php`:
```js
// Sound notification
function newmail_notifier_sound() {
    var src = rcmail.assets_path('static.php/plugins/newmail_notifier/sound');

    (new Audio(src + '.mp3')).play()
        .catch(function () {
            // fallback to the wav format
            (new Audio(src + '.wav')).play();
        });
}
```
Only issue with this is that every "problematic" plugin would need to declare paths in this way.

Some more context; I use Apache 2.4 (Windows) and so I'm making use of `.htaccess` in the root folder of Roundcube to make `public_html` the actual root, for example:
```apacheconf
<IfModule mod_rewrite.c>
Options +SymLinksIfOwnerMatch
RewriteEngine On
RewriteCond %{REQUEST_URI} !^public_html
RewriteRule ^(.*)$ public_html/$1 [L]
</IfModule>
```

If I replace `SymLinksIfOwnerMatch` with `FollowSymlinks`, I don't get 403 errors and don't need to change anything in the plugins, it works fine, but I'm afraid I may be downgrading the level of security of the folder by doing that since it's still bypassing `static.php`. 

It's unclear to me what is the best approach is in this situation, despite thinking that **all** resources should be served by `static.php` though.

Thoughts?

**Comment by alecpl** [comment]
> Some more context; I use Apache 2.4 (Windows) and so I'm making use of `.htaccess` in the root folder of Roundcube to make `public_html` the actual root, for example:

Why? Why not to point Apache to `public_html`. That's the intention.

> It's unclear to me what is the best approach is in this situation, despite thinking that **all** resources should be served by `static.php` though.

All resources should be served by `static.php`.

**Comment by alecpl** [comment]
I have no issues with newmail_notifier plugin, when using both `https://webmail.domain.tld` and `https://domain.tld/webmail`, so it must be something with your HTTP server setup.

**Comment by alecpl** [comment]
I just used nginx and added this to my config:
```
location ~ /roundcube/.* {
    rewrite "^/roundcube/(.*)$" /$1;
}
```
So, it's not as it should be done, but for this test I think it's fine. The links were `https://domain.tld/roundcube/static.php/plugins/...` as expected.
